src/python/platforms/android/adb.py

```diff
@@ -371,16 +372,33 @@ def recreate_gce_device():
   stop_gce_device()
 
   # Delete all existing images.
-  image_dir = cvd_dir
-  for image_file_path in glob.glob(os.path.join(image_dir, '*.img')):
-    shell.remove_file(image_file_path)
-
-  # Restore images from backup.
-  backup_image_dir = os.path.join(cvd_dir, 'backup')
-  for image_filename in os.listdir(backup_image_dir):
-    image_src = os.path.join(backup_image_dir, image_filename)
-    image_dest = os.path.join(image_dir, image_filename)
-    shell.copy_file(image_src, image_dest)
+  cvd_address = get_gce_address()
+  rm_cmd = 'rm -rf {cvd_dir}/*'.format(cvd_dir=cvd_dir)
+  if cvd_address:
+    rm_cmd = 'ssh %s %s' % (cvd_address, rm_cmd)
+  execute_command(rm_cmd, timeout=RECOVERY_CMD_TIMEOUT)
+
+  # Copy and Combine cvd host package and OTA images.
+  image_directory = environment.get_value('IMAGES_DIR')
+  for image_filename in os.listdir(image_directory):
+    image_src = os.path.join(image_directory, image_filename)
+    if cvd_address:
+      if image_src.endswith('.tar.gz'):
```

Comment:
There's a lot of duplication here. Let's consolidate this into the one block for both cuttlefish and non-cuttlefish, by making helpers as described above.

Reply:
Done